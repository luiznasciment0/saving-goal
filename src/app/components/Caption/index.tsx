import Typography from 'app/components/Typography'

interface CaptionProps {
  children: React.ReactNode
}

const Caption = ({ children }: CaptionProps) => (
  <Typography as="p" size="xsmall" weight="light" color="secondaryBlack">
    {children}
  </Typography>
)

export default Caption
