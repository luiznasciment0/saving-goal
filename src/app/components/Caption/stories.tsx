import { Story, Meta } from '@storybook/react/types-6-0'
import Caption from '.'

export default {
  title: 'Caption',
  component: Caption,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => <Caption>{args.children}</Caption>

Default.args = {
  children: 'Teste'
}
