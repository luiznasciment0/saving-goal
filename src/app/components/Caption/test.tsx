import { render, screen } from '@testing-library/react'

import Caption from '.'

describe('<Caption />', () => {
  it('should render the Caption and its styles', () => {
    const { container } = render(<Caption>Test</Caption>)

    expect(screen.getByText(/test/i)).toHaveStyle({
      'font-size': '1.2rem',
      'line-height': '1.6rem',
      'font-weight': '400',
      'font-family': `'Work Sans'`,
      color: '#1c1e1f'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
