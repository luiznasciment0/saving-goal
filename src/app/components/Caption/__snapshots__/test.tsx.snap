// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Caption /> should render the Caption and its styles 1`] = `
.c0 {
  font-size: 1.2rem;
  line-height: 1.6rem;
  font-family: 'Work Sans';
  color: #1c1e1f;
  font-weight: 400;
}

<p
  class="c0"
  color="secondaryBlack"
>
  Test
</p>
`
